### ISSUE TITLE
How can I run Celery Beat/Worker on PC2 while my Django application is hosted at PC1

### ISSUE THREAD

**Issue by MuhammadBilal1** [celery/celery#8558]
### Current scenario:
**PC1:**
1- Django Project (multiple applications)
2- Celery Beat
3- Celery Worker
4- Redis
5- PostgreSQL

Celery configurations in `settings.py` file of Django project: :arrow_down: 
```
# Celery Configuration Options
CELERY_BROKER_URL = 'redis://localhost:6379'
CELERY_RESULT_BACKEND = 'redis://localhost:6379'
CELERY_ACCEPT_CONTENT = ['application/json']
CELERY_TASK_SERIALIZER = 'json'
CELERY_RESULT_SERIALIZER = 'json'
CELERY_TIMEZONE = 'Asia/Karachi'
CELERY_TASK_TRACK_STARTED = True
CELERY_TASK_TIME_LIMIT = 30 * 60
CELERY_BEAT_SCHEDULER = 'django_celery_beat.schedulers:DatabaseScheduler'
CELERY_ENABLE_UTC = False
```

Celery configurations in Django project `celery.py`: :arrow_down: 
```
from __future__ import absolute_import
import os
from django.conf import settings
from celery import Celery

# set the default Django settings module for the 'celery' program.
os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'myproject1.settings')
app = Celery('myproject1')
app.config_from_object('django.conf:settings', namespace='CELERY')
app.autodiscover_tasks(lambda: settings.INSTALLED_APPS)
app.autodiscover_tasks()
```
I run my Celery Beat/Worker with below commands and it works fine:
```celery -A myproject1 beat -l debug```
```celery -A myproject1 worker -l debug```

### Planned scenario:
My plan is distribute the load from PC1 from PC2, so I am trying to move my Celery Beat/Worker & Redis from PC1 to PC2.
So far, I configured my Redis over PC2 and it's working fine. I am able to access the Redis from PC1 by putting the IP of my PC2 in Redis connection information.
**Note:** I don't want to keep a copy of my source code on PC2. I just want Celery over there and all my source code should be on my PC1.

**PC1:**
1- Django Project
2- PostgreSQL

Celery configurations in `settings.py` file of Django project: :arrow_down: 
```
# Celery Configuration Options
CELERY_BROKER_URL = 'redis://default:password@192.168.1.4:6379'
CELERY_RESULT_BACKEND = 'redis://default:password@192.168.1.4:6379'
CELERY_ACCEPT_CONTENT = ['application/json']
CELERY_TASK_SERIALIZER = 'json'
CELERY_RESULT_SERIALIZER = 'json'
CELERY_TIMEZONE = 'Asia/Karachi'
CELERY_TASK_TRACK_STARTED = True
CELERY_TASK_TIME_LIMIT = 30 * 60
CELERY_BEAT_SCHEDULER = 'django_celery_beat.schedulers:DatabaseScheduler'
CELERY_ENABLE_UTC = False
```
Celery configurations in Django project `celery.py`: :arrow_down: 
```
from __future__ import absolute_import
import os
from django.conf import settings
from celery import Celery

# set the default Django settings module for the 'celery' program.
os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'myproject1.settings')
app = Celery('myproject1')
app.config_from_object('django.conf:settings', namespace='CELERY')
app.autodiscover_tasks(lambda: settings.INSTALLED_APPS)
app.autodiscover_tasks()
```

**PC2:**
1- Celery Beat
2- Celery Worker
3- Redis

Upon running the celery beat `celery beat -l debug` and `celery worker -l debug` beat and worker are running, but when I schedule a task from my Django application, I am getting a message that **worker got an unregistered task**.
**How can I link my celery to my Django application, so it gets the tasks of my Django application and works?**

**Comment by jlucas91** [comment]
Your stated goal of not maintaining a copy of your source code on PC2 is not possible - the celery worker instance needs a copy of the task code to process the tasks it picks up from the broker. 

Copy the codebase to PC2 and everything should work as expected (as most celery deployments use this model).

**Comment by MuhammadBilal1** [comment]
> Your stated goal of not maintaining a copy of your source code on PC2 is not possible - the celery worker instance needs a copy of the task code to process the tasks it picks up from the broker.
> 
> Copy the codebase to PC2 and everything should work as expected (as most celery deployments use this model).

@jlucas91 Thank you for your response.
Actually, my project is still in the development phase and use of celery worker/beat is too much, so it would be very tough for me to copy the codebase on PC2 after every change, so I found out a different way out.

**The solution I implemented:**
Instead of copying the codebase, I just shared (mounted) the directory of my codebase from PC1 to PC2 and ran the celery (beat/worker) from the mounted directory.

I used [NFS server](https://ubuntu.com/server/docs/service-nfs) on PC1 and used [NFS client](https://ubuntu.com/server/docs/service-nfs#:~:text=NFS%20Client%20Configuration) over PC2 to mount the volume of my codebase.
